[kaiyi27] iP

kaiyi27 opened the pull request:
### ArtemisPro

>  “If you spend too much time thinking about a thing, you’ll never get it done.” – Bruce Lee ([source](https://dansilvestre.com/productivity-quotes))

ArtemisPro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER FAST_ to use 
- Efficient for fast typer

All you need to do is,

1. download it from [here](https://github.com/kaiyi27/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing todo tasks
- [x] Managing deadlines
- [x] Managing events
- [ ] Reminders (coming soon)

---

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

kaiyi27 committed:
Level 0. Rename, Greet, Exit

kaiyi27 committed:
Level 1. Echo

kaiyi27 committed:
Level 2. Add, List

kaiyi27 committed:
Level 3. Mark as Done

kaiyi27 committed:
Level 4. ToDos, Events, Deadlines

kaiyi27 committed:
Level 4. ToDos, Events, Deadlines with Inheritance

kaiyi27 committed:
Automated Text UI Testing

kaiyi27 committed:
Level 5. Handle Errors

kaiyi27 committed:
Level 6. Delete

kaiyi27 committed:
Space bar

kaiyi27 committed:
Level 7. Save

kaiyi27 committed:
Add exception to handle the situation of the data file being corrupted

kaiyi27 committed:
Merge branch 'branch-Level-7' into master

* branch-Level-7:
  Add exception to handle the situation of the data file being corrupted
  Level 7. Save

kaiyi27 committed:
Level 8. Dates and Times

kaiyi27 committed:
Merge branch 'branch-Level-8' into master

* branch-Level-8:
  Level 8. Dates and Times

kaiyi27 committed:
Add Parser, Storage, TaskList and Ui

kaiyi27 committed:
Add Parser, Storage, TaskList, Ui classes

kaiyi27 committed:
Use More OOP

kaiyi27 committed:
Rename duke to artemis

kaiyi27 committed:
Organize the classes into a package

kaiyi27 committed:
Merge commit add-gradle-support into master

kaiyi27 committed:
Add data.txt, ParserTest and TodoTest

kaiyi27 committed:
Add DeadlineTest and EventTest

kaiyi27 committed:
Add details and testcase to ParserTest

kaiyi27 committed:
Add StorageTest

kaiyi27 committed:
Add ip.jar into jarfile

kaiyi27 committed:
Add javadocs to Artemis class

kaiyi27 committed:
Add javadocs to ArtemisException and Deadline

kaiyi27 committed:
Add javadocs to Event, Parser and Storage

kaiyi27 committed:
Add javadocs to Task, TaskList, Todo and Ui

kaiyi27 committed:
Modify the coding style

kaiyi27 committed:
Add new line to the end of the file

kaiyi27 committed:
Add Find task method in Parser and Ui

kaiyi27 committed:
Update artemis.txt

kaiyi27 committed:
Merge branch 'branch-A-CodingStandard' into master

* branch-A-CodingStandard:
  Add new line to the end of the file
  Modify the coding style

kaiyi27 committed:
Merge branch 'branch-A-JavaDoc' into master

* branch-A-JavaDoc:
  Add javadocs to Task, TaskList, Todo and Ui
  Add javadocs to Event, Parser and Storage
  Add javadocs to ArtemisException and Deadline
  Add javadocs to Artemis class

# Conflicts:
#	src/main/java/com/example/artemis/ArtemisException.java
#	src/main/java/com/example/artemis/Deadline.java
#	src/main/java/com/example/artemis/Event.java

kaiyi27 committed:
Resolve the merge conflict

kaiyi27 committed:
Merge branch 'branch-Level-9' into master

* branch-Level-9:
  Update artemis.txt
  Add Find task method in Parser and Ui

# Conflicts:
#	src/main/java/com/example/artemis/Parser.java
#	src/main/java/com/example/artemis/Ui.java

kaiyi27 committed:
Solve the merge conflict of branch-Level-9

kaiyi27 committed:
Modify the indention

kaiyi27 committed:
Modify the variable name in Event

kaiyi27 committed:
Modify the source file path

kaiyi27 committed:
Add DateTimeException for Parser

kaiyi27 committed:
Add config checkstyle folder

kaiyi27 committed:
Modify all the violation of java coding style

kaiyi27 committed:
Merge branch 'branch-A-CheckStyle' into master

* branch-A-CheckStyle:
  Modify all the violation of java coding style

kaiyi27 committed:
Create a method start for GUI

kaiyi27 committed:
Add GUI implementation to build.gradle

kaiyi27 committed:
Create a Launcher class

kaiyi27 committed:
Turn the constant variable to static final

kaiyi27 committed:
Add AnchorPane to store scrollpane, textfield and button

kaiyi27 committed:
Add scroll and interact feature to the GUI

kaiyi27 committed:
Modify according to javaFX tutorial 3

kaiyi27 committed:
Use FXML to rebuild the scene

kaiyi27 committed:
Modify the method to return a String

kaiyi27 committed:
Modify the GUI

kaiyi27 committed:
Modify the javadocs

kaiyi27 committed:
Merge branch 'branch-Level-10' into master

* branch-Level-10:
  Modify the javadocs
  Modify the GUI
  Modify the method to return a String
  Use FXML to rebuild the scene
  Modify according to javaFX tutorial 3
  Add scroll and interact feature to the GUI
  Add AnchorPane to store scrollpane, textfield and button
  Turn the constant variable to static final
  Create a Launcher class
  Add GUI implementation to build.gradle
  Create a method start for GUI

kaiyi27 committed:
Add background color to dialog

kaiyi27 committed:
Turn the image view to circle

kaiyi27 committed:
Delete unused variable

kaiyi27 committed:
Set the gui resizable to false

kaiyi27 committed:
Modify text field

kaiyi27 committed:
Create jarfile and add into jarfile

gavin331 commented on src/main/java/com/example/artemis/Parser.java:
```diff
@@ -0,0 +1,184 @@
+            }
+
+            String description = tokens[0].replace("event ", "").trim();
+            String[] fromTo = tokens[1].split("/to");
```
Should this be in plural form instead of singular?

gavin331 commented on src/main/java/com/example/artemis/Ui.java:
```diff
@@ -0,0 +1,159 @@
+     * @param message The error message to be displayed.
+     * @return A formatted error message.
+     */
+    public String showError(String message) {
```
I like how you used line wrapping to make the code more readable although it may not have exceeded the word limit.

gavin331 commented on src/test/java/com/example/artemis/ParserTest.java:
```diff
@@ -0,0 +1,38 @@
+import org.junit.jupiter.api.Test;
+public class ParserTest {
+    @Test
+    public void testHandleTodoTask() {
```
Perhaps these test method names can be modified to follow the three part format? Eg. featureUnderTest_testScenario_expectedBehavior()

gavin331 commented on src/test/java/com/example/artemis/StorageTest.java:
```diff
@@ -0,0 +1,73 @@
+
+import java.io.File;
+import java.util.ArrayList;
+
```
Is the blank line necessary here?

gavin331 reviewed (commented):
Overall the code looks good to me except for some minor coding standard violations. Consider modifying the names of the test methods to follow the standards (Three parts format). Eg. featureUnderTest_testScenario_expectedBehavior()

kaiyi27 committed:
Implement assertion checks in all classes

Assert is neccessary in some class to define assumptions about the program state

Added assertion statements to all classes within the project to enforce specific conditions and ensure program correctness. Assertions are used to validate assumptions about the program state at runtime. By incorporating these assertions, we aim to catch any unexpected behaviors or errors early in development, thus improving code reliability and maintainability.

- Modified all classes to include assertion statements
- Verified that assertions cover critical conditions and invariants
- Conducted testing to ensure assertions trigger appropriately under different scenarios

kaiyi27 committed:
Merge pull request #1 from kaiyi27/branch-A-Assertions

Implement assertion checks in all classes

kaiyi27 committed:
Merge branch 'master' of https://github.com/kaiyi27/ip

* 'master' of https://github.com/kaiyi27/ip:
  Implement assertion checks in all classes

kaiyi27 committed:
Improve code quality for all classes

The codebase lacks consistent coding conventions, leading to varying styles and readability issues across classes.

Inconsistent coding conventions can make the codebase difficult to maintain, understand, and collaborate on.

Standardize coding conventions across all classes, including naming conventions, formatting, and documentation style.

Standardizing coding conventions promotes readability, maintainability, and consistency throughout the codebase. Consistent conventions make it easier for developers to understand and modify code, as they know what to expect in terms of structure and style.

kaiyi27 committed:
Merge pull request #2 from kaiyi27/branch-A-CodeQuality

Improve code quality for all classes

kaiyi27 committed:
Merge branch 'master' of https://github.com/kaiyi27/ip

* 'master' of https://github.com/kaiyi27/ip:
  Improve code quality for all classes

kaiyi27 committed:
Add method to handle duplicate task

kaiyi27 committed:
Merge branch 'branch-C-DetectDuplicates' into master

* branch-C-DetectDuplicates:
  Add method to handle duplicate task

kaiyi27 committed:
Add some details to handle duplicate task

SamuelZero1 commented on src/main/java/Duke.java:
Nice use of OOP and nice java style(e.g have fixed the wildcard import problem)

kaiyi27 committed:
Modify the user and bot png

kaiyi27 committed:
Add Ui.png into docs file

kaiyi27 committed:
Add user guide into README.md

kaiyi27 committed:
Modify some minor detail in README.md

kaiyi27 committed:
Update README.md

kaiyi27 committed:
Delete old README.md

kaiyi27 committed:
Create the new jar file

kaiyi27 committed:
Change data
